.vib/nginx-ingress-controller/ginkgo/nginx_ingress_controller_test.go

```diff
@@ -49,6 +49,14 @@ var _ = Describe("NGINX Ingress Controller:", func() {
 				panic(fmt.Sprintf("There was an error checking whether the testing ingress had an IP assigned: %q", err))
 			}
 			Expect(hasIP).To(BeTrue())
+
+			resolves, err = retry("resolvesToDeployment", 6, 20*time.Second, func() (bool, error) {
+				return resolvesToDeployment(ctx, "http://"+ingressHost)
+			})
+			if err != nil {
+				panic(fmt.Sprintf("There was an error resolving the ingress host: %q", err))
+			}
+			Expect(resolves).To(BeTrue())
```

Comment:
This is OK, but I think it does not belong here. You should move it to the `the host resolves to the testing deployment` test case below